docs/src/man/basics.md

```diff
@@ -0,0 +1,2170 @@
+  996 │   995     31  female      1  own      little           NA
+  997 │   996     40  male        3  own      little           little
+  998 │   997     38  male        2  own      little           NA              ⋯
+  999 │   998     23  male        2  free     little           little
+ 1000 │   999     27  male        2  own      moderate         moderate
+                                                  4 columns and 985 rows omitted
+```
+
+We will get the vector stored in our DataFrame without copying it:
+
+```jldoctest dataframe
+julia> german.Age 
+1000-element Vector{Int64}:
+ 67
+ 22
+ 49
+ 45
+ 53
+ 35
+ 53
+ 35
+ 61
+ 28
+  ⋮
+ 34
+ 23
+ 30
+ 50
+ 31
+ 40
+ 38
+ 23
+ 27
+```
+
+We will get the vector stored in our data frame without copying it using String:
+
+```jldoctest dataframe
+julia> german."Sex" 
+1000-element PooledArrays.PooledVector{String, UInt32, Vector{UInt32}}:
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ ⋮
+ "male"
+ "male"
+ "male"
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+
+julia> german[:, 3] # note that this creates a copy
+1000-element PooledArrays.PooledVector{String, UInt32, Vector{UInt32}}:
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ ⋮
+ "male"
+ "male"
+ "male"
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+```
+
+To get two columns as a `DataFrame`, we can index as follows:
+
+```jldoctest dataframe
+julia> german[:, 1:2]
+1000×2 DataFrame
+  Row │ id     Age
+      │ Int64  Int64
+──────┼──────────────
+    1 │     0     67
+    2 │     1     22
+    3 │     2     49
+    4 │     3     45
+    5 │     4     53
+    6 │     5     35
+    7 │     6     53
+    8 │     7     35
+  ⋮   │   ⋮      ⋮
+  994 │   993     30
+  995 │   994     50
+  996 │   995     31
+  997 │   996     40
+  998 │   997     38
+  999 │   998     23
+ 1000 │   999     27
+     985 rows omitted
+
+julia> german[:, [3]] # here we have created a single column data frame
+1000×1 DataFrame
+  Row │ Sex
+      │ String
+──────┼────────
+    1 │ male
+    2 │ female
+    3 │ male
+    4 │ male
+    5 │ male
+    6 │ male
+    7 │ male
+    8 │ male
+  ⋮   │   ⋮
+  994 │ male
+  995 │ male
+  996 │ female
+  997 │ male
+  998 │ male
+  999 │ male
+ 1000 │ male
+985 rows omitted
+```
+
+Compare this with accessing the vector contained in column `3`:
+
+```jldoctest dataframe
+julia> german[:, 3]
+1000-element PooledArrays.PooledVector{String, UInt32, Vector{UInt32}}:
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ ⋮
+ "male"
+ "male"
+ "male"
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+```
+
+We can get a single cell of a data frame with the same syntax to get a cell of a matrix:
+
+```jldoctest dataframe
+julia> german[4, 4]
+2
+```
+
+To get a new `DataFrame` that is subset of rows and columns do e.g.:
```

Comment:
again - several lines above you gave a very similar example `german[1:5, [:Sex, :Age]]`. What is the logic of order of examples here?

Reply:
In that example, I was using Column names `:Sex` and `:Age`, and I thought I should give an example that how to access columns using an index.

Reply:
And this is a good idea, but if you want to compare two things they should be put closer together and an appropriate comment explaining the difference.

Reply:
Now i have added a comment.